packages/modules/vehicles/common/vehicle_utils.py

```diff
@@ -0,0 +1,242 @@
+from modules.common.configurable_vehicle import ConfigurableVehicle
+from control import data
+from modules.common.component_state import CarState
+
+
+DEF_LOGFILTER = '__'
+
+log = logging.getLogger(__name__)
+
+
+# logUtils - filtered logging
+# if logFilter is defined log filtered messages only otherwise log normally on INFO level
+class logUtils:
+
+    def __init__(self, filter: Optional[str] = None):
+        if filter:
+            self.logFilter = filter
+        else:
+            self.logFilter = DEF_LOGFILTER
+
+    def set_logFilter(self, filter: str = ''):
+        self.logFilter = filter
+
+    # get logFilter via mqtt
+    def get_logFilter(self, vnum: str) -> str:
+        mq = mqttUtils(self)
+        try:
+            msg = mq.mqtt_get_topic("openWB/vehicle/" + vnum + "/soc_module/config", "localhost")
+            config = loads(str(msg.payload, 'utf-8'))
+            if 'logFilter' in config['configuration']:
+                self.logFilter = config['configuration']['logFilter']
+                if self.logFilter is None:
+                    self.logFilter = DEF_LOGFILTER
+            else:
+                self.logFilter = DEF_LOGFILTER
+        except Exception as e:
+            log.exception("get_logFilter mqtt error: %s", e)
+            self.logFilter = DEF_LOGFILTER
+        self.infoLog('Q', 'get_logFilter: logFilter=' + self.logFilter)
+        return self.logFilter
+
+    def infoLog(self, filter: str, txt: str):
+        if self.logFilter != DEF_LOGFILTER:
+            if filter in self.logFilter or 'A' in self.logFilter:
+                log.info('(' + filter + '): ' + txt)
+        else:
+            log.info(txt + ', filter=' + filter + ', logFilter=' + self.logFilter)
+
+
+# mqtt section
+class mqttUtils:
+
+    def __init__(self, lu: Optional[logUtils] = None):
+        if lu:
+            self.lu = lu
+        else:
+            self.lu = logUtils()
+
+    # write rsp. publish vehicle configuration
+    def write_config(self, topic: str, config: dict):
+        try:
+            self.lu.infoLog('i', "write_config:\n" + dumps(config, ensure_ascii=False, indent=4))
+            Pub().pub(topic, config)
+        except Exception as e:
+            log.exception('Token mqtt write exception ' + str(e))
+
+    # publish a single message to 1883 broker
+    def pub(self, topic: str, payload: str):
+        try:
+            self.lu.infoLog('q', "pub: topic=" + topic + ', payload=' + payload)
+            client = mqtt.Client()
+            client.connect("localhost", 1883, 60)
+            ret, mid = client.publish(topic, payload)
+            if ret != 0:
+                self.lu.infoLog('q', "pub: topic=" + topic +
+                                ', payload=' + payload +
+                                ', ret=' + str(ret) +
+                                ', mid=' + str(mid))
+            client.disconnect()
+        except Exception as e:
+            log.exception("pub mqtt error: %s", e)
```

Comment:
Warum braucht es dafür extra Code? In pub.py gibt es die Funktion Pub().pub() und pub_single(). Geloggt werden topic und payload im mqtt.log.

Reply:
Danke für den Hinweis, das sehe ich mir noch mal an.

Reply:
Wird im nächsten commit entfernt.